src/TestFramework/TestFramework/Attributes/TestMethod/TaskRunTestMethodAttribute.cs

```diff
@@ -0,0 +1,85 @@
+/// </list>
+/// </para>
+/// <para>
+/// For best results, test methods should observe the cancellation token and cancel cooperatively.
+/// If the test method does not handle cancellation properly, the task may continue running after the timeout,
+/// which can still lead to issues, but the test runner will not block waiting for it to complete.
+/// </para>
+/// </remarks>
+[AttributeUsage(AttributeTargets.Method)]
+public class TaskRunTestMethodAttribute : TestMethodAttribute
+{
+    private readonly TestMethodAttribute? _testMethodAttribute;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TaskRunTestMethodAttribute"/> class.
+    /// </summary>
+    public TaskRunTestMethodAttribute([CallerFilePath] string callerFilePath = "", [CallerLineNumber] int callerLineNumber = -1)
+        : base(callerFilePath, callerLineNumber)
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TaskRunTestMethodAttribute"/> class.
+    /// This constructor is intended to be called by test class attributes to wrap an existing test method attribute.
+    /// </summary>
+    /// <param name="testMethodAttribute">The wrapped test method.</param>
+    public TaskRunTestMethodAttribute(TestMethodAttribute testMethodAttribute)
+        : base(testMethodAttribute.DeclaringFilePath, testMethodAttribute.DeclaringLineNumber ?? -1)
+        => _testMethodAttribute = testMethodAttribute;
+
+    /// <summary>
+    /// Executes a test method by wrapping it in a <see cref="Task.Run"/> to allow timeout handling.
+    /// </summary>
+    /// <param name="testMethod">The test method to execute.</param>
+    /// <returns>An array of TestResult objects that represent the outcome(s) of the test.</returns>
+    public override async Task<TestResult[]> ExecuteAsync(ITestMethod testMethod)
+    {
+        if (_testMethodAttribute is not null)
+        {
+            return await ExecuteWithTaskRunAsync(() => _testMethodAttribute.ExecuteAsync(testMethod)).ConfigureAwait(false);
+        }
+
+        return await ExecuteWithTaskRunAsync(() => testMethod.InvokeAsync(null)).ConfigureAwait(false);
+    }
+
+    private static async Task<TestResult[]> ExecuteWithTaskRunAsync(Func<Task<TestResult>> executeFunc)
+    {
+        // Run the test method in Task.Run so that we can stop awaiting it on timeout
+        // while allowing it to complete in the background
+        Task<TestResult> testTask = Task.Run(executeFunc);
+
+        TestResult result = await testTask.ConfigureAwait(false);
+        return [result];
+    }
+
+    private static async Task<TestResult[]> ExecuteWithTaskRunAsync(Func<Task<TestResult[]>> executeFunc)
+    {
+        // Run the test method in Task.Run so that we can stop awaiting it on timeout
+        // while allowing it to complete in the background
+        Task<TestResult[]> testTask = Task.Run(executeFunc);
```

Comment:
If STA with specified in runsettings, we are breaking it here because threadpool threads are not STA and we need to create our own custom thread.

I'm also not sure if we should simply just always create our custom thread and never use the threadpool here.